server/src/main/java/io/spine/server/entity/storage/EntityRecordStorage.java

```diff
@@ -294,48 +273,38 @@ private static boolean hasNoIds(Query<?, ?> query) {
                     .isEmpty();
     }
 
-    private static <I> boolean hasNoLifecycleCols(Query<I, ?> query) {
-        Subject<I, ?> subject = query.subject();
-        ImmutableList<? extends QueryPredicate<?>> predicates = subject.predicates();
-        for (QueryPredicate<?> predicate : predicates) {
-            boolean hasColumn = predicate.parameters()
-                                         .stream()
-                                         .anyMatch((param) -> isLifecycleColumn(param.column()));
-            if (hasColumn) {
-                return false;
-            }
-        }
-        return true;
-    }
-
     private RecordQuery<I, EntityRecord> onlyActive(RecordQuery<I, EntityRecord> query) {
         RecordQuery<I, EntityRecord> result = query;
         if (hasNoIds(query) && hasNoLifecycleCols(query)) {
             RecordQueryBuilder<I, EntityRecord> builder = query.toBuilder();
-            if(hasArchivedColumn) {
-                builder.where(archived.asRecordColumn(Boolean.class)).is(false);
-            }
-            if(hasDeletedColumn) {
-                builder.where(deleted.asRecordColumn(Boolean.class)).is(false);
-            }
+            builder.where(archived).is(false)
+                   .where(deleted).is(false);
             result = builder.build();
         }
         return result;
     }
 
-    private static boolean isLifecycleColumn(Column<?, ?> column) {
-        ColumnName name = column.name();
-        return name.equals(archived.columnName()) || name.equals(deleted.columnName());
-    }
-
-    private FindActiveEntites.Builder<I, S> findActiveEntities() {
-        return FindActiveEntites.newBuilder(idType(), stateType(),
-                                            hasArchivedColumn, hasDeletedColumn);
+    private static <I> boolean hasNoLifecycleCols(Query<I, EntityRecord> query) {
+        Subject<I, ?> subject = query.subject();
+        ImmutableList<? extends QueryPredicate<?>> predicates = subject.predicates();
+        for (QueryPredicate<?> predicate : predicates) {
+            boolean hasColumn =
+                    predicate.parameters()
+                             .stream()
+                             .anyMatch((param) -> isLifecycleColumn(param.column()));
```

Comment:
What if we use `map(Param::column).anyMatch(::isLifecycleColumn)` instead?

Reply:
@yuri-sergiichuk I tried your version of this piece and was looking at it for quite some time :) To my taste, a single `anyMatch()` is easier to understand than `map()` + `anyMatch()` with method references as their arguments. So, thanks for the suggestion, but maybe next time.